src/pages/integrations/gift-cards/confirm-card-purchase/confirm-card-purchase.ts

```diff
@@ -455,6 +503,8 @@ export class ConfirmCardPurchasePage extends ConfirmPage {
       buyerSelectedTransactionCurrency: COIN,
       cardName: this.cardConfig.name
     };
+
+    this.logGiftCardPurchaseEvent(false, COIN, this.tx.giftData);
```

Comment:
This is producing a typescript error, since when the function runs `this.tx.` is not yet defined.
(The strange thing is that the error is not shown by console, I had to follow the code step by step with `debugger`)
This causes the gift card data to never be brought.
Maybe what you really want to send as a parameter is: `dataSrc`
Instead of `this.tx.giftData` (?